streams/src/test/java/org/apache/kafka/streams/processor/internals/NamedTopologyTest.java

```diff
@@ -112,15 +103,15 @@ public void shouldReturnTopologyByName() {
         final NamedTopology topology2 = builder2.buildNamedTopology(props);
         final NamedTopology topology3 = builder3.buildNamedTopology(props);
         streams = new KafkaStreamsNamedTopologyWrapper(asList(topology1, topology2, topology3), props, clientSupplier);
-        assertThat(streams.getTopologyByName("topology-1"), equalTo(topology1));
-        assertThat(streams.getTopologyByName("topology-2"), equalTo(topology2));
-        assertThat(streams.getTopologyByName("topology-3"), equalTo(topology3));
+        assertThat(streams.getTopologyByName("topology-1").get(), equalTo(topology1));
+        assertThat(streams.getTopologyByName("topology-2").get(), equalTo(topology2));
+        assertThat(streams.getTopologyByName("topology-3").get(), equalTo(topology3));
     }
 
     @Test
-    public void shouldThrowIllegalArgumentWhenLookingUpNonExistentTopologyByName() {
+    public void shouldReturnEmptyWhenLookingUpNonExistentTopologyByName() {
```

Comment:
We changed the behavior to return an empty Optional rather than throw, as users may want to use this API to determine whether the given named topology is known or not